### PR TITLE
update calculix to 2.17

### PR DESCRIPTION
**Pull request by galanom**
This are the ebuilds and patches for current version, 2.17, of calculix-ccx and calculix-cgx.
I used the master branch files of version 2.10 as a template. I did not alter any DEPEND variable.
It works for me and I hope it might be useful for others.

### CODE REVIEW

**Comment by waebbl** [comment]
Thanks for your contribution! Long time since I last checked that package... Glad someone noticed the updated package :)

The code in the ebuilds is quite old and from the time, when I began writing ebuilds. As such it could use some improvements.
Would you be willing to participate in a review of the code and improve it to adhere current ebuild standard?
It's ok, if you don't want to do this, I then do it myself.

**Comment by galanom** [comment]
Hello,
I'll be happy to help if I can, but keep in mind I've never written an ebuild.
If there is a resource describing the changes you need, please let me know.

**Comment by waebbl** [comment]
@galanom That's ok. I will start a review process and we will together go through the code.
Resources which might be useful are the devmanual: https://devmanual.gentoo.org/index.html which contains a host of information about writing ebuilds. More technical and theoretical and probably less useful at this time is the PMS: https://dev.gentoo.org/~ulm/pms/head/pms.html

Of some importance, however, is, before commiting your changes, please run `repoman -dx manifest` to update the Manifest and `repoman -dx full` to check the ebuilds. Also please run `pkgcheck scan .` in the ebuild directory.

File: sci-libs/calculix-ccx/calculix-ccx-2.17.ebuild [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
```
Comment (waebbl) [inline sci-libs/calculix-ccx/calculix-ccx-2.17.ebuild:2]: Please update the last copyright year. Should look like `1999-2021`

File: sci-libs/calculix-ccx/calculix-ccx-2.17.ebuild [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+EAPI=7
+
+inherit eutils toolchain-funcs flag-o-matic fortran-2
```
Comment (waebbl) [inline sci-libs/calculix-ccx/calculix-ccx-2.17.ebuild:6]: Is eutils still needed?

File: sci-libs/calculix-ccx/calculix-ccx-2.17.ebuild [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+MY_P=ccx_${PV/_/}
+
+DESCRIPTION="A Free Software Three-Dimensional Structural Finite Element Program"
+HOMEPAGE="http://www.calculix.de/"
```
Comment (waebbl) [inline sci-libs/calculix-ccx/calculix-ccx-2.17.ebuild:11]: Could you please check if https is working for the homepage?

File: sci-libs/calculix-ccx/calculix-ccx-2.17.ebuild [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+SRC_URI="
+	http://www.dhondt.de/${MY_P}.src.tar.bz2
+	doc? ( http://www.dhondt.de/${MY_P}.ps.tar.bz2 )
+	examples? ( http://www.dhondt.de/${MY_P}.test.tar.bz2 )"
```
Comment (waebbl) [inline sci-libs/calculix-ccx/calculix-ccx-2.17.ebuild:15]: Same here

File: sci-libs/calculix-ccx/calculix-ccx-2.17.ebuild [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+IUSE="arpack doc examples openmp threads"
+
+RDEPEND="
+	arpack? ( >=sci-libs/arpack-3.1.3 )
+	>=sci-libs/spooles-2.2[threads=]
+	virtual/lapack
+	virtual/blas"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig
+	doc? ( app-text/ghostscript-gpl )"
```
Comment (waebbl) [inline sci-libs/calculix-ccx/calculix-ccx-2.17.ebuild:29]: `virtual/pkgconfig` and the dependency for the `doc` use flag are both build time dependencies. They now belong to the `BDEPEND` variable.
The `DEPEND` line should just read `DEPED="${RDEPEND}"` here

File: sci-libs/calculix-ccx/calculix-ccx-2.17.ebuild [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+S=${WORKDIR}/CalculiX/${MY_P}/src
```
Comment (waebbl) [inline sci-libs/calculix-ccx/calculix-ccx-2.17.ebuild:31]: Nowadays, the `S` entry goes in the description block above. Please move it just beyond the `SRC_URI` variable.

File: sci-libs/calculix-cgx/calculix-cgx-2.17.ebuild [hunk 7]
```diff
@@ -0,0 +1,55 @@
+# Copyright 1999-2018 Gentoo Authors
```
Comment (waebbl) [inline sci-libs/calculix-cgx/calculix-cgx-2.17.ebuild:1]: Please adjust copyright year here too

File: sci-libs/calculix-cgx/calculix-cgx-2.17.ebuild [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+MY_P=cgx_${PV}
+
+DESCRIPTION="A Free Software Three-Dimensional Structural Finite Element Program"
+HOMEPAGE="http://www.calculix.de/"
+SRC_URI="http://www.dhondt.de/${MY_P}.all.tar.bz2
+	doc? ( http://www.dhondt.de/${MY_P}.pdf )"
```
Comment (waebbl) [inline sci-libs/calculix-cgx/calculix-cgx-2.17.ebuild:11]: Same as with calculix-ccx: please check for https availability

File: sci-libs/calculix-cgx/calculix-cgx-2.17.ebuild [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+RDEPEND="media-libs/mesa:=
+	>=media-libs/freeglut-1.0:=
+	virtual/opengl:=
+	x11-libs/libX11:=
+	x11-libs/libXmu:=
+	x11-libs/libXi:=
+	x11-libs/libXext:=
+	x11-libs/libXt:=
+	x11-libs/libSM:=
+	x11-libs/libICE:="
```
Comment (waebbl) [inline sci-libs/calculix-cgx/calculix-cgx-2.17.ebuild:27]: None of these dependencies actually needs the slot operator (`:=`). Please remove it from all of them.

File: sci-libs/calculix-cgx/calculix-cgx-2.17.ebuild [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+DEPEND="${RDEPEND}
+	doc? ( app-text/ghostscript-gpl )"
```
Comment (waebbl) [inline sci-libs/calculix-cgx/calculix-cgx-2.17.ebuild:29]: Same as with calculix-ccx: the `doc` line belongs to `BDEPEND`.

File: sci-libs/calculix-cgx/calculix-cgx-2.17.ebuild [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+
+S=${WORKDIR}/CalculiX/${MY_P}/src/
```
Comment (waebbl) [inline sci-libs/calculix-cgx/calculix-cgx-2.17.ebuild:31]: Please move to the description, homepage block above.

**Review by waebbl — CHANGES_REQUESTED** [review]
Could you please try to fix these?

**Comment by galanom** [comment]
Hello,
Thanks for taking the time to help me, and I'm sorry for late reply!
Regarding the changes requested:
a) I see that https does not seem to work, therefore no changes made to URLs
b) All RDEPENDs of calculix-cgx are not slotted, therefore I removed the ":=" from all of them, not just libXext, libXt, libSM, libICE
I believe new commit is automatically visible, please tell me if you need further changes.

**Comment by waebbl** [comment]
> Thanks for taking the time to help me, and I'm sorry for late reply!

You're welcome. Help with package maintainance on Gentoo systems is always needed :)
Don't worry about long reply times, there's no need to hurry. I find it more important to get some quality than to get it done fast.

> Regarding the changes requested:
> a) I see that https does not seem to work, therefore no changes made to URLs

That's ok. I just didn't know. We can't do anything about it anyway.

> b) All RDEPENDs of calculix-cgx are not slotted, therefore I removed the ":=" from all of them, not just libXext, libXt, libSM, libICE

It's perfect for the dependencies used in the ebuild.
But note, that not only the slot is important, but also subslots, i.e. slots in the form of e.g. `0/27`, i.e. two slot identifiers divided by slash sign. If a package has just one slot, but uses subslots, then the slot operator is also needed to properly trigger automatic rebuild of revdeps. It doesn't apply here though.

> I believe new commit is automatically visible, please tell me if you need further changes.

The ebuilds look good to me. I'm going to test them and merge if they work.

**Comment by waebbl** [comment]
There's an issue with the calculix-cgx ebuild with USE=doc. The pdf file doesn't get unpacked, because the unpacker doesn't know how to handle PDF files. For this, a src_unpack function needs to be implemented. I'm going to do this myself. Anything else is working.
Thanks for your contribution.

**Pull request merged**